src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py

```diff
@@ -448,16 +447,15 @@ def values_statement(output: List[Attribute], data: List[Row]) -> str:
             for value, data_type in zip(row, data_types)
         ]
         rows.append(LEFT_PARENTHESIS + COMMA.join(cells) + RIGHT_PARENTHESIS)
-    query_source = (
-        VALUES
+
+    query = (
+        SELECT
+        + COMMA.join([f"{DOLLAR}{i+1}{AS}{c}" for i, c in enumerate(names)])
+        + FROM
+        + VALUES
         + COMMA.join(rows)
-        + AS
-        + table_name
-        + LEFT_PARENTHESIS
-        + COMMA.join(names)
```

Comment:
I'm curious why we chose the previous way.

Reply:
Not sure if it's a randomized choice, or a purposeful one. The [VALUES documentation](https://docs.snowflake.com/en/sql-reference/constructs/values.html) doesn't say they have difference except that the one we chose to use can have alias.
I think the column aliases have a bug for quotes but it's hard to change (would-be break change).